src/tidy.zig

```diff
@@ -69,6 +81,117 @@ fn tidy_long_line(file: SourceFile) !void {
     }
 }
 
+/// `tigerbeetle/cli.zig`'s `parse_args` holds the function line coiunt record as of July 26, 2024
+/// with 366 lines. As we trim our functions, make sure to update this constant; tidy will
+/// error if you do not.
+const maximum_allowable_function_line_count = 366;
+
+fn tidy_long_functions(
+    file: SourceFile,
+    maximum_allowable_function_line_count_needs_updating: *bool,
```

Comment:
We generally try to avoid this pattern of passing out parameters for bools and such, as it makes following the control flow harder, and makes it easier to accidentally step into some aliasing nastiness. 

I'd maybe do something like this:

```
fn longest_function(file: SourceFile) struct { line_count: u32, first_line } {
}
```

and then the caller can do control flow based on `maximum_allowable_function_line_count_needs_updating` (which would get a reduced scope, and could become a shorter name because of that)

Reply:
Thanks, this also made the logic way clearer!

Reply:
It's useful to reflect _why_ it became clearer. I think the main principle here is "centralize control flow": https://github.com/tigerbeetle/tigerbeetle/commit/47af705b7aebdb6361ae1a15092131fb93bba322